spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -99,6 +109,35 @@ public UnicastSendingMessageHandler(String host, int port) {
 		super(host, port);
 		this.mapper.setLengthCheck(false);
 		this.mapper.setAcknowledge(false);
+		this.socketExpression = null;
+	}
+
+	/**
+	 * Basic constructor using external datagram socket for sending outgoing data.
+	 * @param socketExpression socket expression string describing external socket
+	 * @since 4.3
+	 */
+	public UnicastSendingMessageHandler(String socketExpression) {
+		super("", 0);
+		Assert.hasText(socketExpression);
+		Assert.state(!this.acknowledge, "socketExpression cannot be used together with acknowledge");
+		this.mapper.setLengthCheck(false);
+		this.mapper.setAcknowledge(false);
+		this.socketExpression = this.EXPRESSION_PARSER.parseExpression(socketExpression);
```

Comment:
:smile:
Didn't changed this - lots of usages and I think it's out of this PR scope.